templates/util_test.go

```diff
@@ -0,0 +1,68 @@
+package templates
+
+import (
+	"testing"
+	"text/template"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindTemplates(t *testing.T) {
+	tests := []struct {
+		name     string
+		tmplName string
+		tmplText string
+		expected []string
+	}{{
+		name:     "inline template",
+		tmplName: "foo",
+		tmplText: `{{ range .Alerts }}{{ end }}`,
+		expected: []string{"foo"},
+	}, {
+		name:     "nested template",
+		tmplName: "foo",
+		tmplText: `{{ define "bar" }}{{ range .Alerts }}{{ end }}{{ end }}`,
+		expected: []string{"bar"},
+	}, {
+		name:     "inline call to nested template",
+		tmplName: "foo",
+		tmplText: `{{ define "bar" }}{{ range .Alerts }}{{ end }}{{ end }}{{ template "bar" . }}`,
+		expected: []string{"foo"},
+	}, {
+		name:     "inline call to nested template in if",
+		tmplName: "foo",
+		tmplText: `{{ define "bar" }}{{ range . }}The labels are {{.Labels }}{{ end }}{{ end }}{{ if len .Alerts }}{{ template "bar" .Alerts }}{{ end }}`,
+		expected: []string{"foo"},
+	}, {
+		name:     "inline call to nested template in range",
+		tmplName: "foo",
+		tmplText: `{{ define "bar" }}The labels are {{.Labels }}{{ end }}{{ range .Alerts }}{{ template "bar" . }}{{ end }}`,
+		expected: []string{"foo"},
+	}, {
+		name:     "inline call to nested template in with",
+		tmplName: "foo",
+		tmplText: `{{ define "bar" }}{{ range . }}The labels are {{.Labels }}{{ end }}{{ end }}{{ with .Alerts }}{{ template "bar" . }}{{ end }}`,
+		expected: []string{"foo"},
+	}, {
+		name:     "inline call to nested template that calls other nested template in if",
+		tmplName: "foo",
+		tmplText: `{{ define "baz" }}{{ range . }}The labels are {{.Labels }}{{ end }}{{ end }}{{ define "bar" }}{{ if len .Alerts }}{{ template "baz" .Alerts }}{{ end }}{{ end }}`,
+		expected: []string{"bar"},
+	}, {
+		name:     "inline call to nested template that calls other nested template in range",
+		tmplName: "foo",
+		tmplText: `{{ define "baz" }}The labels are {{.Labels }}{{ end }}{{ define "bar" }}{{ with .Alerts }}{{ template "baz" . }}{{ end }}{{ end }}`,
+		expected: []string{"bar"},
```

Comment:
I see the tests only assert a single item in the resulting slice. Can you add coverage for cases when there are multiple items?